daskms/experimental/zarr/tests/test_zarr.py

```diff
@@ -407,3 +408,27 @@ def test_xds_from_zarr_assert_on_empty_store(tmp_path_factory, ms):
 
     with pytest.raises(UnknownStoreTypeError, match="Unable to infer table type"):
         xds_from_zarr(path)
+
+
+@pytest.mark.skipif(xarray is None, reason="depends on xarray")
+def test_xarray_reading_daskms_written_dataset(ms, tmp_path_factory):
+    store = tmp_path_factory.mktemp("zarr_store")
+
+    datasets = xds_from_ms(ms)
+
+    for i, ds in enumerate(datasets):
+        chunks = ds.chunks
+        row = sum(chunks["row"])
+        chan = sum(chunks["chan"])
+        corr = sum(chunks["corr"])
+
+        datasets[i] = ds.assign_coords(
+            row=np.arange(row), chan=np.arange(chan), corr=np.arange(corr)
+        )
+
+    path = store / "test.zarr"
+    dask.compute(xds_to_zarr(datasets, path, consolidated=True))
+
+    for i, mem_ds in enumerate(datasets):
+        ds = xarray.open_zarr(path / "MAIN" / f"MAIN_{i}")
```

Comment:
@david-macmahon With this change, the standard xarray interface works (although complains about not being able to read consolidated metadata). I'll investigate this a bit further

```bash
============================================ warnings summary =============================================
daskms/experimental/zarr/tests/test_zarr.py::test_xarray_reading_daskms_written_dataset
daskms/experimental/zarr/tests/test_zarr.py::test_xarray_reading_daskms_written_dataset
daskms/experimental/zarr/tests/test_zarr.py::test_xarray_reading_daskms_written_dataset
  /home/simon/code/dask-ms/daskms/experimental/zarr/tests/test_zarr.py:433: RuntimeWarning: Failed to open Zarr store with consolidated metadata, but successfully read with non-consolidated metadata. This is typically much slower for opening a dataset. To silence this warning, consider:
  1. Consolidating metadata in this existing store with zarr.consolidate_metadata().
  2. Explicitly setting consolidated=False, to avoid trying to read consolidate metadata, or
  3. Explicitly setting consolidated=True, to raise an error in this case instead of falling back to try reading non-consolidated metadata.
    ds = xarray.open_zarr(path / "MAIN" / f"MAIN_{i}")

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
============================== 1 passed, 30 deselected, 3 warnings in 0.28s ===============================
```